### PR TITLE
Cat 535 Performance: Risk profiler batch needs to be more efficient

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderEvent.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+/**
+ * A COPY of the custody-api class
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderEvent.java]: understand this is a copy.. maybe jackson's `JsonSubType`s would have been a nicer way to have modelled this originally

File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/services/NomisService.java [hunk 2]
```diff
@@ -33,10 +33,10 @@
     };
     private static final ParameterizedTypeReference<List<Map>> OFFENDERS = new ParameterizedTypeReference<>() {
     };
-    private static final String[] ESCAPE_LIST_ALERT_TYPES = {"XER", "XEL"};
+    public static final List<String> ESCAPE_LIST_ALERT_TYPES = Arrays.asList("XER", "XEL");
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/services/NomisService.java]: meh.. could have used? `List.of(

File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/services/NomisService.java [hunk 3]
```diff
@@ -85,9 +85,8 @@ public void evictSocListAlertsCache(final String nomsId) {
         return getCandidates(uri);
     }
 
-    //@Cacheable("incident")
-    // TODO caching disabled for now as the initial query is too slow
-    public List<IncidentCase> getIncidents(@NotNull final String nomsId, @NotNull final List<String> incidentTypes, final List<String> participationRoles) {
+    @Cacheable(value = "incident", key = "#p0")
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/services/NomisService.java]: is it safe to just cache using the first parameter?

File: src/test/java/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public void testIncidents() {
+
+        when(nomisService.getPartiesOfIncident(INCIDENT_1)).thenReturn(Arrays.asList(OFFENDER_1, OFFENDER_2));
+
+        service.eventListener("{ \"Message\": \"{ \\\"eventType\\\":\\\"INCIDENT-CHANGED-CASES\\\", \\\"incidentCaseId\\\":" + INCIDENT_1 + " }\"}");
+
+        verify(nomisService).evictIncidentsCache(OFFENDER_1);
+        verify(nomisService).evictIncidentsCache(OFFENDER_2);
+//        verify(pollPrisonersService).pollPrisoner(OFFENDER_1);
+//        verify(pollPrisonersService).pollPrisoner(OFFENDER_2);
```
Comment (andrewrlee) [inline src/test/java/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListenerTest.java]: could maybe delete these comments?

File: build.gradle [hunk 5]
```diff
@@ -95,6 +95,9 @@ dependencies {
     implementation 'redis.clients:jedis:3.1.0'
 
     implementation("org.springframework.cloud:spring-cloud-starter-aws-messaging:2.2.0.RELEASE")
+    implementation 'org.springframework:spring-jms'
+    implementation 'com.amazonaws:amazon-sqs-java-messaging-lib:1.0.8'
+    implementation 'org.apache.camel:camel-spring-boot:2.23.2'
```
Comment (andrewrlee) [inline build.gradle]: is camel being used here?

File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderEvent.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+    private Long sanctionSeq;
+    private Long movementSeq;
+    private Long imprisonmentStatusSeq;
+    private Long assessmentSeq;
+
+    private Long alertSeq;
+    private LocalDateTime alertDateTime;
+    private String alertType;
+    private String alertCode;
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderEvent.java]: Most of these values are not needed

File: build.gradle (hunk 5 above)
Comment (steverendell) [reply build.gradle]: this was upgraded so diff is misleading

File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/services/NomisService.java (hunk 3 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/services/NomisService.java]: yes, the others are config

File: src/test/java/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListenerTest.java (hunk 4 above)
Comment (steverendell) [reply src/test/java/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListenerTest.java]: OK

**Pull request merged**